server/sonar-server/src/test/java/org/sonar/server/authentication/event/AuthenticationEventImplTest.java

```diff
@@ -238,6 +238,69 @@ public void failure_logs_X_Forwarded_For_header_from_request_and_supports_multip
     verifyLog("login failure [cause|Boom!][method|EXTERNAL][provider|REALM|bar][IP|1.2.3.4|2.3.4.5,6.5.4.3,9.5.6.7,6.3.2.4][login|foo]");
   }
 
+  @Test
+  public void logout_fails_with_NPE_if_request_is_null() {
+    logTester.setLevel(LoggerLevel.INFO);
+    expectedException.expect(NullPointerException.class);
+    expectedException.expectMessage("request can't be null");
+
+    underTest.logout(null, "foo", null);
+  }
+
+  @Test
+  public void logout_does_not_interact_with_request_if_log_level_is_above_DEBUG() {
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    logTester.setLevel(LoggerLevel.INFO);
+
+    underTest.logout(request, "foo", null);
+
+    verifyZeroInteractions(request);
+  }
+
+  @Test
+  public void logout_creates_DEBUG_log_with_empty_login_if_login_argument_is_null() {
+    underTest.logout(mockRequest(), null, null);
+
+    verifyLog("logout [error|][IP||][login|]");
+  }
+
+  @Test
+  public void logout_creates_DEBUG_log_with_login() {
+    underTest.logout(mockRequest(), "foo", null);
+
+    verifyLog("logout [error|][IP||][login|foo]");
+  }
+
+  @Test
+  public void logout_creates_DEBUG_log_with_error() {
+    underTest.logout(mockRequest(), "foo", "bad token");
+
+    verifyLog("logout [error|bad token][IP||][login|foo]");
```

Comment:
This is out of scope of this ticket => Log format has been validated by nemops, ask @sns-seb  and @ehartmann for more info